### DISCUSSION TITLE
How to access production database?

### DISCUSSION THREAD

**Discussion by pleirb** [Tecnativa/doodba-copier-template#478] [Q&A] [answered]
I have answered the copier's questions and I have answered yes to the option of exposing the database on port 5432, but still, I cannot connect to the database through the domain. Is it necessary to enable something else?

**Comment by pleirb** [comment]
I have everything configured according to the documentation, however I cannot access the production database through port 5432
And my bringfik detects this error that is surely related:

![imagen](https://github.com/Tecnativa/doodba-copier-template/assets/2092475/a3a66288-ee92-46cf-a720-39e4b4cdc115)

**Reply by pedrobaeza** [reply]
It seems to only work with Traefik 3

**Reply by pleirb** [reply]
Yes, I figured, I tried to do some tests with #476 but I didn't have any success either, in fact I get even more errors. Is there an example configuration for Traefik 3 that we can test with?

**Reply by pedrobaeza** [reply]
cc @josep-tecnativa

**Reply by pleirb** [reply, marked as answer]
> It seems to only work with Traefik 3

You were right, I can make it work with Traefik 3, but I had to make modifications to the service db tags in prod.yaml
For the main domain add:
`traefik.tcp.routers.project-16-0-prod-database-0.tls: true`
And replace:
```
traefik.tcp.routers.project-16-0-prod-database-0.rule:
  Host(`domain.com`)
```
By:
```
traefik.tcp.routers.project-16-0-prod-database-0.rule:
  HostSNI(`domain.com`)
```
**AND**
For subdomains that redirect to the main domain add:
traefik.tcp.routers.project-16-0-prod-database-1.tls: true
And replace:
```
traefik.tcp.routers.project-16-0-prod-database-1.rule:
  Host(`subdomain1.domain.com`, `subdomain2.domain.com`, `subdomain3.domain.com`, `subdomain4.domain.com`)
```
By:
```
traefik.tcp.routers.project-16-0-prod-database-1.rule:
  HostSNI(`subdomain1.domain.com`) || HostSNI(`subdomain2.domain.com`) || HostSNI(`subdomain3.domain.com`) || HostSNI(`subdomain4.domain.com`)
```
Then you have to make sure that the following entrypoint is added to the traefik.yml configuration file:
```
  tcp:
    address: ":5432"
```
And it is also necessary to map the port in the docker-compose.yaml of the traefik service, which of course must be configured with the image in version 3.0
```
    ports:
      - "80:80"
      - "443:443"
      - "5432:5432"
```

I hope this information is helpful to be able to modify the copier assistant and make it functional. On the other hand, the reverse proxy examples for versions 2 and 3 of Traekif will be missing in the documentation (which are still absent)

**Comment by pleirb** [comment]
This is the status of the HTTP Router with Traefik 3

![imagen](https://github.com/Tecnativa/doodba-copier-template/assets/2092475/ab01b431-c449-4257-aaba-a3a641b6c37b)

And this is the status of the TCP Router with Traefik 3

![imagen](https://github.com/Tecnativa/doodba-copier-template/assets/2092475/753cbf48-6df0-4cd1-abcb-c6908954f7f0)

Related case #477

**Reply by josep-tecnativa** [reply]
As far I can see, your errors come from not having the Traefik 3 rules. This is because you probably did a `copier update`, but since there hasn't been a new release, you don't get the option to choose Traefik 3.

While waiting for a new release, you can use the branch from this PR https://github.com/Tecnativa/doodba-copier-template/pull/479 to generate labels compatible with Traefik 3.

How can you do this? Either update your projects with `copier update --vcs-ref add-external-db-access-feature --trust`, or you can do `copier copy --vcs-ref add-external-db-access-feature --trust gh:Tecnativa/doodba-copier-template your-project-folder`.

Additionally, this PR adds the feature of connecting to the database externally. Take a look at it. Try as I mentioned and the Host rules should work for you, because from what I see, you have Traefik 2 labels, and that's why you're getting an error.

**Reply by josep-tecnativa** [reply]
Correction: The new release is already available; I hadn't noticed. Have you selected Traefik 3 to ensure the rules are generated correctly?

**Reply by pleirb** [reply]
> Corrección: El nuevo lanzamiento ya está disponible; no me había dado cuenta. Ha seleccionado Traefik 3 para asegurarse de que las reglas se generan correctamente?

I have used the new release to test the Traefik 3 option and updating an existing template. The only tags that are not generated correctly is for the exposed database. On the other hand, although Traefik does not show errors without the database being exposed, I am experiencing problems with the video call of the discuss (chat) module. I would rule out meeting if it is because I use too many domains

**Reply by josep-tecnativa** [reply]
Exposed database rules are not ready yet. I'm working on this PR https://github.com/Tecnativa/doodba-copier-template/pull/479, which I haven't been able to dedicate much time to yet, but I hope it will be ready soon.

**Reply by pleirb** [reply]
> Exposed database rules are not ready yet. I'm working on this PR #479, which I haven't been able to dedicate much time to yet, but I hope it will be ready soon.

Great, yes, on my part I was able to make it work by changing Host to HostSNI and configuring entrypoint on port 5432
